src/target.d

```diff
@@ -82,6 +83,8 @@ struct Target
         c_long_doublesize = realsize;
         if (global.params.is64bit && global.params.isWindows)
             c_long_doublesize = 8;
+
+        cppExceptions = global.params.dwarfeh || (global.params.is64bit && global.params.isLinux);
```

Comment:
This condition looks like hack, IMO. C++ exception implementation doesn't depends on OS or CPU. It depends on C++ ABI.
Lets say that we support some C++ features for some kinds of C++ ABIs and lets reflect that in compiler and library code. As far as I see, current implementation should works with C++ ABI v1. Lets reflect it in our code: add predefined version `CXXABI_ITANIUM_V1` for example, add special flag into global.params: `isCppItanium`, `isCppMars` or `isCppMSVC` and add flag to compiler in future.
After that lets choose C++ mangling and some other C++ features depending on the ABI flag and reflect it in the docs (explicitly write what C++ features we support for the C++ Itanium ABI, what features for DMC, what features for MSVC.